key.core/src/main/java/de/uka/ilkd/key/nparser/builder/ExpressionBuilder.java

```diff
@@ -375,16 +394,47 @@ public Object visitStrong_arith_term_1(KeYParser.Strong_arith_term_1Context ctx)
         return last;
     }
 
+    @Override
+    public Object visitEmptyset(KeYParser.EmptysetContext ctx) {
+        var op = services.getTypeConverter().getLocSetLDT().getEmpty();
+        return updateOrigin(getTermFactory().createTerm(op), ctx);
+    }
+
     @Override
     public Object visitStrong_arith_term_2(KeYParser.Strong_arith_term_2Context ctx) {
-        Term termL = accept(ctx.a);
-        if (ctx.b == null) {
-            return termL;
+        if (ctx.b.isEmpty()) { // fast path
+            return accept(ctx.a);
         }
 
-        Term termR = accept(ctx.b);
-        String opName = ctx.SLASH() != null ? "div" : "mod";
-        return binaryLDTSpecificTerm(ctx, opName, termL, termR);
+        List<Term> termL = mapOf(ctx.b);
+        // List<String> opName = ctx.op.stream().map(it -> it.getType()== KeYLexer.PERCENT ? "mod" :
+        // "div").collect(Collectors.toList());
+
+        Term term = accept(ctx.a);
+        var sort = term.sort();
+        if (sort == null) {
+            semanticError(ctx, "No sort for term '%s'", term);
+        }
+
+        var ldt = services.getTypeConverter().getLDTFor(sort);
+
+        if (ldt == null) {
+            // falling back to integer ldt (for instance for untyped schema variables)
+            ldt = services.getTypeConverter().getIntegerLDT();
+        }
+
+        assert ctx.op.size() == ctx.b.size();
+
+        for (int i = 0; i < termL.size(); i++) {
+            var opName = ctx.op.get(i).getType() == KeYLexer.PERCENT ? "mod" : "div";
+            Function op = ldt.getFunctionFor(opName, services);
```

Comment:
Is division hence also possible for floats and doubles thanks to this?

Reply:
It does a lookup after the corresponding LDT, and request the function of it. 

It only depends, whether the LDT answers for the given sort.